onnxruntime/core/framework/sparse_utils.cc

```diff
@@ -0,0 +1,500 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/sparse_utils.h"
+#include "core/common/status.h"
+#include "core/framework/tensor.h"
+#include "core/framework/data_types_internal.h"
+#include "core/framework/data_transfer_manager.h"
+#include "core/framework/sparse_tensor.h"
+#include "core/framework/sparse_cooformat_rep.h"
+#include "core/framework/sparse_csrcformat_rep.h"
+
+#include "core/util/math_cpuonly.h"
+
+namespace onnxruntime {
+namespace sparse_utils {
+
+// Copy element
+using CopyElementFunc = void (*)(void* dest, const void* src, int64_t dest_index, int64_t src_index);
+
+template <typename T>
+inline void CopyElement(void* dst, const void* src, int64_t dst_index, int64_t src_index) {
+  reinterpret_cast<T*>(dst)[dst_index] = reinterpret_cast<const T*>(src)[src_index];
+}
+
+void CopyString(void* dst, const void* src, int64_t dst_index, int64_t src_index) {
```

Comment:
>CopyString [](http://example.com/codeflow?start=5&length=10)

is this equivalent to `CopyElement<std::string>`?

Reply:
yes

Reply:
the follow up question is whether it's necessary to define a separate `CopyString()`